### PR TITLE
chore: test icon management

### LINKED ISSUE #7034: Add unit tests for adding, removing, and accessing icons
### Description

Part of #2105 

Add unit tests for the new APIs for custom icons we want to add to blocks:

* addIcon
* removeIcon
* getIcon
* getIcons
* hasIcon

### PR DESCRIPTION
**Pull request by BeksOmega**
## The basics

- [X] I branched from develop
- [X] My pull request is against develop
- [X] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)
- [X] I ran `npm run format` and `npm run lint`

### Resolves

Fixes #7034 

### Proposed Changes

Adds unit tests for adding, removing, and accessing icons.

### Reason for Changes

Tests are good :D

### Test Coverage

This is tests silly!

### Documentation

N/A

### Additional Information

N/A

### CODE REVIEW

File: tests/mocha/block_test.js [hunk 1]
```diff
@@ -1335,12 +1341,139 @@ suite('Blocks', function() {
       }
     });
   });
-  suite('Icon Management', function() {
-    suite('Bubbles and Collapsing', function() {
+
+  suite('icon management', function() {
```
Comment (rachel-fenichel) [inline tests/mocha/block_test.js:1345]: Nit: match capitalization for the rest of the tests.

File: tests/mocha/block_test.js [hunk 2]
```diff
@@ -1335,12 +1341,139 @@ suite('Blocks', function() {
...
+      getType() {return 'A';}
+    }
+
+    class MockIconB {
+      getType() {return 'B';}
+    }
+    
+    suite.skip('adding icons', function() {
+      setup(function() {
+        workspaceTeardown.call(this, this.workspace);
```
Comment (rachel-fenichel) [inline tests/mocha/block_test.js:1356]: Why teardown at the beginning of the test?

File: tests/mocha/block_test.js [hunk 3]
```diff
@@ -1335,12 +1341,139 @@ suite('Blocks', function() {
...
+            'Expected removeIcon to return true');
+        chai.assert.isFalse(
+            this.block.hasIcon('A'),
+            'Expected the icon to be removed');
+      });
+
+      test('removing an icon that does not exist returns false', function() {
+        chai.assert.isFalse(
+            this.block.removeIcon('B'),
+            'Expected removeIcon to return true');
```
Comment (rachel-fenichel) [inline tests/mocha/block_test.js:1415]: This expects false, not true.

File: tests/mocha/block_test.js [hunk 4]
```diff
@@ -1335,12 +1341,139 @@ suite('Blocks', function() {
...
+            iconB,
+            'Expected getIcon to return the icon with the given type');
+      });
+
+      test('if there is no matching icon, getIcon returns null', function() {
+        this.block.addIcon(new MockIconA());
+        chai.assert.isNull(
+            this.block.getIcon('B'),
+            'Expected getIcon to return null if there is no ' +
+            'icon with a missing type');
```
Comment (rachel-fenichel) [inline tests/mocha/block_test.js:1467]: Missing -> matching

File: tests/mocha/block_test.js (hunk 2 above)
Comment (BeksOmega) [reply tests/mocha/block_test.js:1356]: This tears down the old headless workspace and creates a new rendered workspace. Will add comment.

File: tests/mocha/block_test.js (hunk 3 above)
Comment (BeksOmega) [reply tests/mocha/block_test.js:1415]: Thank you for the catch!

**Review by rachel-fenichel — APPROVED** [review]

**Pull request merged**